Remotion/Validation/UnitTests/Implementation/ObjectValidatorFactoryTest.cs

```diff
@@ -15,29 +15,88 @@
 // along with re-motion; if not, see http://www.gnu.org/licenses.
 // 
 using System;
+using Moq;
 using NUnit.Framework;
+using Remotion.Reflection;
+using Remotion.Validation.Implementation;
+using Remotion.Validation.Rules;
+using Remotion.Validation.UnitTests.TestDomain;
+using Remotion.Validation.UnitTests.TestDomain.Validators;
+using Remotion.Validation.UnitTests.TestHelpers;
+using Remotion.Validation.Validators;
 
 namespace Remotion.Validation.UnitTests.Implementation
 {
   [TestFixture]
   public class ObjectValidatorFactoryTest
   {
-    [Ignore("RM-5906")]
     [Test]
-    public void Create ()
+    public void Create_ReturnsValidator ()
     {
+      var fakeCustomerValidator = new FakeCustomerValidator();
+      var validatorFactoryMock = new Mock<Func<ObjectValidationRuleInitializationParameters, IObjectValidator>>();
+      ObjectValidationRuleInitializationParameters initializationParameters = null;
+      validatorFactoryMock
+          .Setup(_ => _(It.IsNotNull<ObjectValidationRuleInitializationParameters>()))
+          .Callback((ObjectValidationRuleInitializationParameters parameters) => { initializationParameters = parameters; })
+          .Returns(fakeCustomerValidator);
+
+      var validatedTypeMock = new Mock<ITypeInformation>();
+      validatedTypeMock
+          .Setup(_ => _.FullName)
+          .Returns(It.IsAny<string>());
+
+      var validationMessageFactoryMock = new Mock<IValidationMessageFactory>(MockBehavior.Strict);
+      validationMessageFactoryMock
+          .Setup(_ => _.CreateValidationMessageForObjectValidator(fakeCustomerValidator, validatedTypeMock.Object))
+          .Returns(new InvariantValidationMessage("test"));
+
+      var result = ObjectValidatorFactory.Create<IObjectValidator>(validatedTypeMock.Object, validatorFactoryMock.Object, validationMessageFactoryMock.Object);
+
+      Assert.That(result, Is.SameAs(fakeCustomerValidator));
+      Assert.That(initializationParameters.ValidationMessage.ToString(), Is.EqualTo("test"));
     }
 
-    [Ignore("RM-5906")]
     [Test]
     public void Create_WithFactoryDelegateReturnsNull_ThrowsInvalidOperationException ()
     {
+      Func<ObjectValidationRuleInitializationParameters, IObjectValidator> factoryDelegate = r => null;
+
+      var validatedTypeMock = new Mock<ITypeInformation>();
+      validatedTypeMock
+          .Setup(_ => _.FullName)
+          .Returns(It.IsAny<string>());
+
+      var validationMessageFactoryMock = new Mock<IValidationMessageFactory>();
+      validationMessageFactoryMock
+          .Setup(_ => _.CreateValidationMessageForObjectValidator(null, validatedTypeMock.Object))
+          .Returns(It.IsAny<ValidationMessage>());
+
+      Assert.That(() => ObjectValidatorFactory.Create<IObjectValidator>(validatedTypeMock.Object, factoryDelegate, validationMessageFactoryMock.Object),
+          Throws.TypeOf<InvalidOperationException>().With.Message.EqualTo("validatorFactory evaluated and returned null."));
```

Comment:
Im used to seeing `Throws.InstanceOf` instead of `TypeOf` from my own code.

I researched and found that instanceOf also includes derivatives, while typeOf does not. Just thought I should tell you as well, in case you were wondering. 😊 (I will probably use TypeOf now instead of instanceOf...)

Reply:
There's no clear precedent in the code base. Both are uses roughly the same. The dicsussion is correct, TypeOf is more precise and the question is, should you care when writing the assertion. For exceptions I think it's not relevant in most cases because you only care about it being a specific base type, not something more specific unless you explicitly want to test for that. This means, InstanceOf makes the test more robust.

For this particular instance, though, I recommend the shorthand `Throws.InvalidOperationException` though (underneath, it uses TypeOf, but nobody is using InvalidOpEx as a basetype)

And please move the delegate in line 75 into its own line, too. If we shart line wrapping, each parameter gets its own line, including the first one.